clients/cobol-lsp-vscode-extension/src/task/CustomBuildTaskTerminal.ts

```diff
@@ -0,0 +1,142 @@
+        }
+
+        return new Promise<void>(async resolve => {
+            try {
+                await this.CompileCobolDocument(zoweExplorerApi, loadedProfile);
+                this.closeEmitter.fire(0);
+                resolve();
+            } catch (error) {
+                return this.handleError(error);
+            }
+        });
+    }
+
+    private async CompileCobolDocument(zoweExplorerApi: ZoweExplorerApi.IApiRegisterClient, loadedProfile: IProfileLoaded) {
+        const jclToSubmit = this.getCompileJobJCL();
+        const jesApi = zoweExplorerApi.getJesApi(loadedProfile);
+
+        this.writeEmitter.fire("Submitting JCL ...\r\n");
+        const submitJob = await jesApi.submitJcl(jclToSubmit);
+        this.writeEmitter.fire(`JOB submitted. JOBID: ${submitJob.jobid} ...\r\n`);
+        this.writeEmitter.fire(`Fetching result ...\r\n`);
+        // TODO : Add a logic to not aggresively look for O/P
+        let job = await jesApi.getJob(submitJob.jobid);
+        do {
+            job = await jesApi.getJob(submitJob.jobid);
+        } while (!job.retcode);
+
+        this.writeEmitter.fire(`JOB completed, return code: ${job.retcode} ...\r\n`);
+        if(`${job.retcode}` !== "JCL ERROR") {
+        const sysPrintSpool = (await jesApi.getSpoolFiles(submitJob.jobname, submitJob.jobid))
+            .filter(file => file.ddname === "SYSPRINT");
+        const sysPrintContent = await jesApi.getSpoolContentById(submitJob.jobname, submitJob.jobid, sysPrintSpool[0].id);
+
+        this.writeEmitter.fire("Analysing output...\r\n");
+
+        this.writeEmitter.fire(sysPrintContent.split("\n")
+            .filter(str => str.match(this.ERROR_REGEX))
+            .map(str => {
+                const match = str.match(this.ERROR_REGEX);
+                return `\x1b[4m\x1b[33m${this.documentText.uri.fsPath}:${match[1]}:7\x1b[24m\x1b[39m -72 - ${match[2]}: ${match[3]}`;
+            })
+            .join("\r\n"));
+            this.storeSysPrint(submitJob, sysPrintContent);
+        }
+    }
+
+    private storeSysPrint(submitJob: any, sysPrintContent: any) {
+        if (this.definition.saveSysprint) {
+            const fileLocation = this.definition.sysPrintLocation ? `${this.definition.sysPrintLocation}` : `${vscode.workspace.workspaceFolders[0].uri.fsPath}${path.sep}SYSPRINT`;
+            if (!fs.existsSync(fileLocation)) {
+                fs.mkdirSync(fileLocation, { recursive: true });
+            }
+            fs.writeFileSync(`${fileLocation}${path.sep}${submitJob.jobid}`, sysPrintContent);
+            this.writeEmitter.fire("\r\n");
+            this.writeEmitter.fire(`SYSPRINT at \x1b[4m\x1b[33m${fileLocation}${path.sep}${submitJob.jobid}\x1b[24m\x1b[39m\r\n`);
+        }
+    }
+
+    private getCompileJobJCL() {
+        let jclToSubmit: string;
+        const jobCard = fs.readFileSync(path.resolve(vscode.workspace.workspaceFolders[0].uri.fsPath, this.definition.jobCard)).toString().trim();
```

Comment:
Synchronous methods block the event loop. Please, check out the asynchronous api, like [here](https://github.com/BroadcomMFD-DevOps/che-che4z-lsp-for-hlasm/pull/65/files#diff-d248ebf776456d5c53fc477c8088460084bcc5674ffaf877a8f7c39a8cb15950R277).

Reply:
removed this after job card is taken as string[]